vault/identity_store_util.go

```diff
@@ -341,14 +341,15 @@ func (i *IdentityStore) upsertEntityInTxn(ctx context.Context, txn *memdb.Txn, e
 			case intErr != nil:
 				return intErr
 			}
+
 			// The entity and aliases will be loaded into memdb and persisted
 			// as a result of the merge so we are done here
 			return nil
 		}
 
 		if strutil.StrListContains(aliasFactors, i.sanitizeName(alias.Name)+alias.MountAccessor) {
 			i.logger.Warn(errDuplicateIdentityName.Error(), "alias_name", alias.Name, "mount_accessor", alias.MountAccessor, "entity_name", entity.Name, "action", "delete one of the duplicate aliases")
-			if !i.disableLowerCasedNames {
+			if !persist && !i.disableLowerCasedNames {
```

Comment:
Should this be an OR, not AND?

Reply:
It should be an AND.

Reply:
OK, so this will then cause memdb updates in this case, whereas before we would not do memdb updates and would simply return the error. Just making sure that's the correct behavior.

Reply:
Thanks for asking! The reason behind this change is just that `disableLowerCasedNames` error is captured only in the startup flow and no where else, and it only makes sense to get this error in that flow. But this function is not restricted to the login flow alone. So adding the guard of `persist` being set to `false` ensures that regular operations doesn't result in this error. This is not related to the core fix in this PR, and is something that I happened to notice.